src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.Session.cs

```diff
@@ -746,11 +753,32 @@ private async Task AddDocumentsWithPotentialConflicts(IEnumerable<Document> docu
             {
                 try
                 {
+                    // Add the RenameSymbolAnnotation to the declarations
+                    foreach (var syntaxReference in _renameLocationSet.Symbol.DeclaringSyntaxReferences)
+                    {
+                        var document = solution.GetDocument(syntaxReference.SyntaxTree);
+                        if (document is null)
+                        {
+                            continue;
+                        }
+
+                        var node = await syntaxReference.GetSyntaxAsync().ConfigureAwait(false);
+                        var root = await document.GetSyntaxRootAsync().ConfigureAwait(false);
+
+                        var newRoot = root.ReplaceNode(
+                            node,
+                            node.WithoutAnnotations(RenameSymbolAnnotation.RenameSymbolKind)
+                                .WithAdditionalAnnotations(RenameSymbolAnnotation.Create(_renameLocationSet.Symbol)));
```

Comment:
Shape here is a bit funky: RenameSymbolAnnotation.Create can return null if the symbol doesn't have locations but I guess that's not possible since the symbol itself *has* locations because we're in this loop itself. This would get flagged if we had nullable enabled for this file; can we write this in some way that won't trip up the compiler? Or justmake RenameSymbolAnnotation.Create() throw instead?

Reply:
(you might have won yourself the "getting to null-annotate ConflictResolver" award. Congrats!)